docs/developing-ray-libraries/developing-a-ray-library.md

```diff
@@ -0,0 +1,411 @@
+---
+title: Developing a Ray Library: A Guide
+weight: 3
+---
+
+The [Ray app](https://myray.app) is not a language-specific debugging app - as long as there's an integration library, it can be used with any language.
+
+If you're interested in developing a Ray library for your language of choice, this document will help guide you through the process.  As an example, we'll be creating an example Ray integration library for Javascript/NodeJS; however, the concepts apply to any language.
+
+Estimated time investment: `1 hour`
+Required Tools: [Typescript](https://github.com/Microsoft/TypeScript), [ESBuild](https://github.com/evanw/esbuild), [ray-proxy](https://github.com/permafrost-dev/ray-proxy)
+Companion Repository: [permafrost-dev/creating-a-ray-integration](https://github.com/permafrost-dev/creating-a-ray-integration)
+
+# Creating a javascript integration for Ray
+
+In this guide, we'll create a new javascript library that communicates with Ray.  There are already comprehensive third-party libraries in this space, such as [node-ray](https://github.com/permafrost-dev/node-ray), but this will serve well as an example.
+
+## Goals
+
+Create a javascript library that communicates with Ray that implements the following methods:
+- `color()`
+- `html()`
+- `ray()`
+- `charles()`
+- `send()`
+- `sendRequest()`
+
+## Getting Started
+
+The [`spatie/ray`](https://github.com/spatie/ray) PHP package should be used as a reference - it is the primary library for Ray, and all new functionality is always added here first.  We'll reference its source code as we write our library.
+
+Create a new directory named `ray-library-reference`, and run:
+
+```bash
+cd ./ray-library-reference
+composer init
+composer require spatie/ray
+```
+
+Next, create a PHP script for testing:
+
+```php
+<?php
+
+// ray-test.php
+
+require_once(__DIR__.'/vendor/autoload.php');
+
+ray('test one')->color('red');
+ray()->html('<strong>this is a bold</string>')->color('blue');
+ray()->send('this is a test');
+```
+
+## Tools
+
+To help determine what payload data is actually being sent to Ray from our test PHP script, we'll use the third-party [`permafrost-dev/ray-proxy`](https://github.com/permafrost-dev/ray-proxy) package to intercept and display all payloads being sent to Ray.
```

Comment:
Are the backticks inside of the link brackets an issue? "[`permafrost-dev/ray-proxy`]" instead of "[permafrost-dev/ray-proxy]"?

Reply:
I'm fairly sure that this isn't a problem. Let's leave it like it is.